### ISSUE TITLE
Strict search in enhanced strategy

### ISSUE THREAD

**Issue by jvwong** [PathwayCommons/pathways-search#69]
The enhanced search strategy attempts to first match a search query by tokenizing then identifying all of them, in no particular order, within a documents  *name:* field. 

The suggestion is to implement a stricter search:  Match the exact string - the sequence of tokens (and spaces).

For example, consider the query ```cell cycle```. A stricter search would first attempt to identify anything containing this exact sequence, for example

- ```Entry to mitosis and the cell cycle and activation of E2F```
- ```The eukaryotic cell cycle does not require MYC```

rather than simply

- ```The cycle that exists in the ring structure is found in a cell```

See PathwayCommons/cpath2/issues/269 for a discussion of the upgraded Lucene syntax.

**Comment by jvwong** [comment]
The [prototype](https://github.com/PathwayCommons/pathways-search/blob/43b393ecca8ad475d4978525addc91560feba1c5/js/search/helpers/queryProcessing.js#L74-L77) leverages Lucene syntax grouping to form multiple sub-queries, each with decreasing specificity.

Consider the example query ```cell cycle```. Then the corresponding groups, each with index field are:

1. Exact match
  - ```(name:cell\ cycle)```
2. Fuzzy edges
  - ```(name:*cell\ cycle*)```
3. All, any-order
  - ```(name:*cell* AND name:*cycle*)```
4. Any
  - ```(name:*cell* OR name:*cycle*)```

with the final query then being

```
(name:cell\ cycle) OR (name:*cell\ cycle*) OR (name:*cell* AND name:*cycle*)  OR (name:*cell* OR name:*cycle*)
```

The algorithm also considers gene symbols (removes name field).

**Comment by jvwong** [comment]
Outstanding limitations

1. Doesn't distinguish (non) biological terms
  - ```pi3 AKT signaling lunch is good``` defaults to ```OR``` over set
  -  enhancement: what are the 'keywords'

**Comment by jvwong** [comment]
implemented in f2bb013a0756c2e9600364a4e6ef08e389361558
